crates/astria-composer/src/collectors/geth.rs

```diff
@@ -241,64 +202,142 @@ impl Geth {
 
                         txs_received_counter.increment(1);
 
-                        match executor_handle
-                            .send_timeout(seq_action, EXECUTOR_SEND_TIMEOUT)
-                            .await
-                        {
-                            Ok(()) => {},
-                            Err(SendTimeoutError::Timeout(_seq_action)) => {
-                                warn!(
-                                    transaction.hash = %tx_hash,
-                                    timeout_ms = EXECUTOR_SEND_TIMEOUT.as_millis(),
-                                    "timed out sending new transaction to executor; dropping tx",
-                                );
-                                txs_dropped_counter.increment(1);
-                            }
-                            Err(SendTimeoutError::Closed(_seq_action)) => {
-                                warn!(
-                                    transaction.hash = %tx_hash,
-                                    "executor channel closed while sending transaction; dropping transaction \
-                                     and exiting event loop"
-                                );
-                                txs_dropped_counter.increment(1);
-                                break Err(eyre!("executor channel closed while sending transaction"));
-                            }
+                        if let Err(err) = executor_send_timeout_handler(
+                            &executor_handle,
+                            seq_action,
+                            tx_hash,
+                            &txs_dropped_counter,
+                        ).await {
+                            break Err(err);
                         }
+
                     } else {
                         break Err(eyre!("geth tx stream ended"));
                     }
                 }
             }
         };
 
-        match &reason {
-            Ok(reason) => {
-                info!(reason, "shutting down");
-            }
-            Err(reason) => {
-                error!(%reason, "shutting down");
-            }
-        };
+        log_exit(&reason);
 
         status.send_modify(|status| status.is_connected = false);
 
         // if the loop exits with an error, we can still proceed with unsubscribing the WSS
         // stream as we could have exited due to an error in sending messages via the executor
         // channel.
+        unsubscribe_wss_stream_handler(&tx_stream).await;
```

Comment:
This is overly descriptive while providing very limited semantic meaning to the reader:
```suggestion
        unsubscribe_from_rollup(&tx_stream).await;
```